### PR TITLE
hotfix(condo): HOTFIX added the ability to use several iOS application configs to send voip pushes via APNS

### PR DESCRIPTION
**Pull request by VKislov**

### CODE REVIEW

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/notification/adapters/appleAdapter.js [hunk 1]
```diff
@@ -216,26 +215,41 @@ class AppleAdapter {
...
-                                ...response,
-                                pushToken: notifications[idx].token,
-                                pushType: pushTypes[notifications[idx].token],
-                            })
-                    )
+        if (!isNull(this.#config) && !isEmpty(notifications)) {
+            const notificationsSortedByAppId = {}
+            for (const notification of notifications) {
+                const appId = notification.appId
+                if (!notificationsSortedByAppId[appId]) notificationsSortedByAppId[appId] = [notification]
```
Comment (SavelevMatthew) [inline apps/condo/domains/notification/adapters/appleAdapter.js:222]: just a code-style, add brackets here...
```
if (cond) something
else something  
```
Not that readable

File: apps/condo/domains/notification/adapters/appleAdapter.js [hunk 2]
```diff
@@ -216,26 +215,41 @@ class AppleAdapter {
...
+                else notificationsSortedByAppId[appId].push(notification)
+            }
+            for (const appId of Object.keys(notificationsSortedByAppId)) {
+                const currentConfig = this.#config[appId]
+                if (!currentConfig) {
+                    logger.info({ msg: 'Unknown appId. Config was not found', appId })
```
Comment (SavelevMatthew) [inline apps/condo/domains/notification/adapters/appleAdapter.js:228]: Maybe logger.error or smth like that?

File: apps/condo/domains/notification/adapters/appleAdapter.js [hunk 3]
```diff
@@ -216,26 +215,41 @@ class AppleAdapter {
...
+                    continue
+                }
+                const currentNotificationsBatch = notificationsSortedByAppId[appId]
+                const app = new AppleMessaging(currentConfig)
```
Comment (SavelevMatthew) [inline apps/condo/domains/notification/adapters/appleAdapter.js:232]: Single connection will be implemented later?

**Review by SavelevMatthew — APPROVED** [review]

File: apps/condo/domains/notification/adapters/appleAdapter.js [hunk 4]
```diff
@@ -216,26 +215,45 @@ class AppleAdapter {
...
-                    appleResult.responses = appleResult.responses.map(
-                        (response, idx) =>
-                            ({
-                                ...response,
-                                pushToken: notifications[idx].token,
-                                pushType: pushTypes[notifications[idx].token],
-                            })
-                    )
+        if (!isNull(this.#config) && !isEmpty(notifications)) {
+            const notificationsSortedByAppId = {}
```
Comment (pahaz) [inline apps/condo/domains/notification/adapters/appleAdapter.js:219]: it looks li groupBy function

File: apps/condo/domains/notification/adapters/appleAdapter.js (hunk 3 above)
Comment (pahaz) [reply apps/condo/domains/notification/adapters/appleAdapter.js:236]: It's better to create this instance inside the constructor

**Pull request merged**